cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -70,59 +70,65 @@ func (nodeIpamController *nodeIPAMController) startNodeIpamControllerWrapper(ini
 func startNodeIpamController(ccmConfig *cloudcontrollerconfig.CompletedConfig, nodeIPAMConfig nodeipamconfig.NodeIPAMControllerConfiguration, ctx genericcontrollermanager.ControllerContext, cloud cloudprovider.Interface) (controller.Interface, bool, error) {
 	var serviceCIDR *net.IPNet
 	var secondaryServiceCIDR *net.IPNet
+	var clusterCIDRs []*net.IPNet
+	var nodeCIDRMaskSizes []int
 
 	// should we start nodeIPAM
 	if !ccmConfig.ComponentConfig.KubeCloudShared.AllocateNodeCIDRs {
 		return nil, false, fmt.Errorf("the AllocateNodeCIDRs is not enabled")
 	}
 
-	// failure: bad cidrs in config
-	clusterCIDRs, dualStack, err := processCIDRs(ccmConfig.ComponentConfig.KubeCloudShared.ClusterCIDR)
-	if err != nil {
-		return nil, false, err
-	}
-
-	// failure: more than one cidr but they are not configured as dual stack
-	if len(clusterCIDRs) > 1 && !dualStack {
-		return nil, false, fmt.Errorf("len of ClusterCIDRs==%v and they are not configured as dual stack (at least one from each IPFamily", len(clusterCIDRs))
-	}
+	// the CloudAllocator ipam does not need service or cluster cidrs since it obtains the data from the cloud directly
+	// for backward compatiblity reasons we don't fail if there are any of those values configured
+	if ipam.CIDRAllocatorType(ccmConfig.ComponentConfig.KubeCloudShared.CIDRAllocatorType) != ipam.CloudAllocatorType {
+		// failure: bad cidrs in config
+		clusterCIDRs, dualStack, err := processCIDRs(ccmConfig.ComponentConfig.KubeCloudShared.ClusterCIDR)
```

Comment:
Ah - we are masking clusterCIDRs I think.

We need to do something like
`cidrs, dualStack, err := processCIDRs(ccmConfig.ComponentConfig.KubeCloudShared.ClusterCIDR)
if err != nil {
			return nil, false, err
		}
clusterCIDRs = cidrs
```